system/jlib/jdebug.cpp

```diff
@@ -1966,6 +1977,73 @@ class CExtendedStats  // Disk network and cpu stats
 
         if (!getInterfaceName(ifname))
             ifname.set("eth0");
+
+#ifdef __linux__
+        // MCK - wish libblkid could do this ...
+        // Another way might also be to look for:
+        //   /sys/block/sd*
+        //   /sys/block/nvme*
+        // and match those with entries in /proc/diskstats
+        StringBuffer cmd("lsblk -o TYPE,MAJ:MIN --pairs");
+        Owned<IPipeProcess> pipe = createPipeProcess();
+        StringBuffer output;
+        if (pipe->run("list disks", cmd, nullptr, false, true))
+        {
+            Owned<ISimpleReadStream> pipeReader = pipe->getOutputStream();
+            readSimpleStream(output, *pipeReader);
+        }
+        int exitcode = pipe->wait();
+        if ( (exitcode == 0) && (output.length() > 0) )
+        {
+            char *line = (char *)malloc(output.length()+1);
+            unsigned i=0;
+            unsigned j=i;
+            while (j < output.length())
+            {
+                if ( (output.charAt(j) == '\n') && (j > i) )
+                {
+                    output.getChars(i, j-1, line);
+                    line[j-i] = '\0';
+                    // line: TYPE="disk" MAJ:MIN="259:0"
+                    if (strstr(line, "TYPE=\"disk\"") != nullptr)
+                    {
+                        char *mptr = strstr(line, "MAJ:MIN=\"");
+                        if (mptr != nullptr)
+                        {
+                            char mstr[sizeof(line)];
+                            strcpy(mstr, &mptr[9]);
+                            size32_t len = strlen(mstr);
+                            for (size32_t k=0; k<len;k++)
+                            {
+                                if (mstr[k] == ':')
+                                    mstr[k] = ' ';
+                                if (mstr[k] == '"')
+                                    mstr[k] = '\0';
+                            }
+                            int majnum, minnum;
+                            unsigned items = sscanf(mstr, "%d%d", &majnum, &minnum);
+                            if (items == 2)
+                            {
+                                if ( ( (IDE_DISK_MAJOR(majnum)) && ((minnum&0x3F)==0) ) ||
+                                     ( (SCSI_DISK_MAJOR(majnum)) && ((minnum&0x0F)==0) ) ||
+                                     ( (OTHER_DISK_MAJOR(majnum)) && ((minnum&0x0F)==0) ) ||
+                                     ((minnum&0x3F)==0) )
+                                {
+                                    unsigned mm = (majnum<<16)+minnum;
+                                    majmin.appendUniq(mm);
+                                }
+                            }
+                        }
+                    }
+                    i = j+1;
+                    j = i;
+                    continue;
+                }
+                j++;
+            }
+            free(line);
+        }
+#endif // __linux__
```

Comment:
picky: I think should format the rest of any function you touch in Allman/ANSI style, i.e. braces on newline, redoing the whole module is a bit much, but I think reasonable to not have mixed stlyes in same function.

Reply:
Everything in CExtendedStats() is the same format ?

Reply:
at minimum I make sure a method/function is same with regularized format.
But if class is smallish very tempting to do class at same time, yes.